lutris/util/timer.py

```diff
@@ -2,20 +2,41 @@
 
 
 class Timer:
+    """Simple Timer class to time code"""
 
-    def start_t(self):
-        self.start = datetime.datetime.now()
+    def __init__(self, format):
+        self.format = format
+        self._start = None
+        self._end = None
+        self.finished = False
 
-    def end_t(self):
-        self.end = datetime.datetime.now()
+    def start(self):
+        self._end = None
+        self._start = datetime.datetime.now()
+        self.finished = False
 
+    def end(self):
+        self._end = datetime.datetime.now()
+        self.finished = True
+
+    @property
     def duration(self):
-        return (self.end - self.start).seconds / 3600
+        if not self._start:
+            result = 0
+
+        elif not self.finished:
+            now = datetime.datetime.now()
+            result = (now - self._start).seconds
 
-    def increment(self, saved_dur):
-        if saved_dur == '':
-            return "%.1f hrs" % self.duration()
+        else:
+            result = (self._end - self._start).seconds
 
-        saved_dur = float(saved_dur.split()[0])
-        new_dur = saved_dur + self.duration()
-        return "%.1f hrs" % new_dur
+        # Retrun time in the specified format
+        if self.format == 'seconds':
```

Comment:
put the conversion in a private helper method. Also this should work as a single if-elif construct with a single return, if I am not mistaken.

Reply:
I can make it do with one return but i just like the expressiveness with current style, is it ok to keep it like this or this is just not pythonic ?